src/03_primitives.jl

```diff
@@ -262,5 +262,6 @@ end
 
 # ----- Base.reverse ---------------------------------------------------------#
 
-# Use ready-made method for AbstractArrays or implement method specific to
-# NullableArrays, possibly for performance purposes?
+function Base.reverse(X::NullableVector, s=1, n=length(X))
+    return(NullableArray(reverse(X.values, s, n), reverse(X.isnull, s, n)))
```

Comment:
Totally unsafe!

```
julia> Y = Any[collect(1:3), collect(4:6), collect(7:9), nothing]
4-element Array{Any,1}:
 [1,2,3]
 [4,5,6]
 [7,8,9]
 nothing

julia> X = NullableArray(Y, Array, Void)
4-element NullableArrays.NullableArray{Array{T,N},1}:
 Nullable([1,2,3])     
 Nullable([4,5,6])     
 Nullable([7,8,9])     
 Nullable{Array{T,N}}()

julia> reverse(X)
ERROR: UndefRefError: access to undefined reference
 in reverse! at array.jl:657
 in reverse1 at none:2
```
Base handles null entries just fine, actually, since it uses `setindex!` which has the behavior you mentioned in your other comment. Here's my hand at writing a safe mutating version out of which to build the pure version:

```julia
function Base.reverse!(X::NullableVector, s=1, n=length(X))
    if isbits(eltype(X)) || !anynull(X)
        reverse!(X.values, s, n)
        reverse!(X.isnull, s, n)
    else
        r = n
        for i in s:div(s+n-1, 2)
            if !X.isnull[i]
                if !X.isnull[r]
                    X.values[i], X.values[r] = X.values[r], X.values[i]
                else
                    X.values[r] = X.values[i]
                end
            else
                if !X.isnull[r]
                    X.values[i] = X.values[r]
                end
            end
            r -= 1
        end
        reverse!(X.isnull, s, n)
    end
    return X
end
```